### PR TITLE
Add FIPS support to new junit-rule test clusters framework

### PR DESCRIPTION
**Pull request by mark-vieira**
This ensures that when running with FIPS enabled via `tests.fips.enabled=true` that test clusters orchestrated via the new rule-based framework are properly configured with a FIPS security provider.

### CODE REVIEW

**Comment by mark-vieira** [comment]
@slobodanadamovic I've fixed the issue that new test clusters weren't setup with FIPS at all. This should sort that out but I'm not sure how to verify it's working correctly. I do see messages like this on node startup but otherwise it "works"?

```
[2023-01-18T02:35:32,435][ERROR][o.e.x.c.s.SSLService     ] [test-cluster-0] unsupported ciphers [[TLS_AES_256_GCM_SHA384, TLS_AES_128_GCM_SHA256, TLS_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384]] were requested but cannot be used in this JVM, however there are supported ciphers that will be used [[TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA, TLS_RSA_WITH_AES_256_GCM_SHA384, TLS_RSA_WITH_AES_128_GCM_SHA256, TLS_RSA_WITH_AES_256_CBC_SHA256, TLS_RSA_WITH_AES_128_CBC_SHA256, TLS_RSA_WITH_AES_256_CBC_SHA, TLS_RSA_WITH_AES_128_CBC_SHA]]. If you are trying to use ciphers with a key length greater than 128 bits on an Oracle JVM, you will need to install the unlimited strength JCE policy files.
```

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/part-1-fips

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Comment by mark-vieira** [comment]
@elasticmachine retest this please

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/part-2

File: test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/FipsEnabledClusterConfigProvider.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    public void apply(LocalClusterSpecBuilder builder) {
+        if (isFipsEnabled()) {
+            builder.configFile(
+                "fips_java.security",
+                Resource.fromClasspath(isOracleJvm() ? "fips/fips_java_oracle.security" : "fips/fips_java.security")
+            )
+                .configFile("fips_java.policy", Resource.fromClasspath("fips/fips_java.policy"))
+                .configFile("cacerts.bcfks", Resource.fromClasspath("fips/cacerts.bcfks"))
+                .systemProperty("java.security.properties", "${ES_PATH_CONF}/fips_java.security")
```
Comment (slobodanadamovic) [inline test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/FipsEnabledClusterConfigProvider.java:24]: I think this property value should have additional `=` char as in [elasticsearch.fips.gradle](https://github.com/elastic/elasticsearch/blob/2ca52e530a1e799361f22925f480455621a2d2f3/build-tools-internal/src/main/groovy/elasticsearch.fips.gradle#L63) file:
```suggestion
                .systemProperty("java.security.properties", "=${ES_PATH_CONF}/fips_java.security")
```

This was intentional in order to get the double equals sign when specifying system property, i.e. it should look like this:  

- `-Djava.security.properties==${ES_PATH_CONF}/fips_java.security`

The `==` means that just the specified security property file will be used.

File: test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/FipsEnabledClusterConfigProvider.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                .systemProperty("java.security.policy", "${ES_PATH_CONF}/fips_java.policy")
```
Comment (slobodanadamovic) [inline test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/FipsEnabledClusterConfigProvider.java:25]: For the same reason as above, we should prefix with `=` char:
```suggestion
                .systemProperty("java.security.policy", "=${ES_PATH_CONF}/fips_java.policy")
```

**Comment by slobodanadamovic** [comment]
> @slobodanadamovic I've fixed the issue that new test clusters weren't setup with FIPS at all. This should sort that out but I'm not sure how to verify it's working correctly. I do see messages like this on node startup but otherwise it "works"?
> 
> ```
> [2023-01-18T02:35:32,435][ERROR][o.e.x.c.s.SSLService     ] [test-cluster-0] unsupported ciphers [[TLS_AES_256_GCM_SHA384, TLS_AES_128_GCM_SHA256, TLS_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384]] were requested but cannot be used in this JVM, however there are supported ciphers that will be used [[TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA, TLS_RSA_WITH_AES_256_GCM_SHA384, TLS_RSA_WITH_AES_128_GCM_SHA256, TLS_RSA_WITH_AES_256_CBC_SHA256, TLS_RSA_WITH_AES_128_CBC_SHA256, TLS_RSA_WITH_AES_256_CBC_SHA, TLS_RSA_WITH_AES_128_CBC_SHA]]. If you are trying to use ciphers with a key length greater than 128 bits on an Oracle JVM, you will need to install the unlimited strength JCE policy files.
> ```

I don't have a good suggestion on how to verify it either, but I think it's fair to assume that it works and applies the FIPS policies if we see this log message.

**Review by slobodanadamovic — APPROVED** [review]
Beside two small suggestions, it LGTM 🚀 

Thank you for implementing this 👍

File: test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/FipsEnabledClusterConfigProvider.java (hunk 1 above)
Comment (mark-vieira) [reply test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/FipsEnabledClusterConfigProvider.java:24]: Thanks. I thought that might have been a typo, however I didn't notice any behavior difference between the two.

**Pull request merged**